### ISSUE TITLE
ChunkUnloadEvent may be asynchronous

### ISSUE THREAD

**Issue by C4K3** [Melvin3000/AntiPVPLog#4]
```
[14:34:23 ERROR]: Could not pass event ChunkUnloadEvent to AntiPVPLog v0.1
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:308) ~[spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.ChunkProviderServer.unloadChunk(ChunkProviderServer.java:313) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.ChunkProviderServer.unloadChunks(ChunkProviderServer.java:288) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.WorldServer.doTick(WorldServer.java:286) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:956) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
        at java.lang.Thread.run(Thread.java:844) [?:?]
Caused by: java.util.ConcurrentModificationException
        at java.util.HashMap$HashIterator.nextNode(HashMap.java:1498) ~[?:?]
        at java.util.HashMap$ValueIterator.next(HashMap.java:1526) ~[?:?]
        at com.Melvin3000.AntiPVPLog.ChunkUnload.onChunkUnload(ChunkUnload.java:22) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor9.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.2-2.jar:git-Spigot-947a8e7-3a91182]
```

Getting that stacktrace occasionally. Basically ChunkUnloadEvent is sometimes fired asynchronously, see https://hub.spigotmc.org/javadocs/bukkit/org/bukkit/event/Event.html#isAsynchronous--

**Comment by C4K3** [comment]
This has only been a problem since 1.13.

It looks like the problem is that it uses a global list of skeletons, and then when unloading a chunk it iterates over those removing those in the chunk. I think we'd need to change it to only iterate over entities in the chunk being unloaded. Is it important that there be a global list? As I read it, we already set a timer to remove entities.

**Comment by Melvin3000** [comment]
Should've been fixed by fd93c45
